corehq/form_processor/utils/general.py

```diff
@@ -9,3 +12,35 @@ def should_use_sql_backend(domain):
     # uncomment when ready for production
     # return USE_SQL_BACKEND.enabled(domain)
     return False
+
+
+def cache_return_value(fn):
+    """
+    Docorator to save the return value of a method or property.
+    Use this where you can't use quickcache or memoize because you can't
+    guarantee a consistent or unique lookup key e.g. forms that get deprecated
```

Comment:
I see. I think another solution would be to incorporate that change into `memoized`

The offending line is
https://github.com/dimagi/dimagi-utils/blob/master/dimagi/utils/decorators/memoized.py#L118-118

I think if we replaced that section with
```
        if self.is_method:
            key = self.get_args_tuple(*args, **kwargs)[1:]
            obj = args[0]
        else:
            key = self.get_args_tuple(*args, **kwargs)
            obj = None
```
it would remove self from the cache key, which is fine because the cache is already per-instance. If that fixed it, would that do everything we want `cache_return_value` to do?

(Also, I'm just trying to be helpful in case we can consolidate the two. If there's more going on than I'm understanding, I'm happy to leave `cache_return_value`.)